goth_tests/domain/ya-provider/test_provider_multi_activity.py

```diff
@@ -159,3 +159,184 @@ async def test_provider_single_simultaneous_activity(
+                    (golem.srv.caps.multi-activity=true)\
+                    (golem.runtime.name=wasmtime))"
+                )
+                .build()
+            )
+
+        async def negotiate_begin(requestor, demand, providers):
+            logger.info(
+                "%s Negotiating with providers",
+                requestor.name,
+            )
+            for provider in providers:
+                await provider.wait_for_offer_subscribed()
+
+            subscription_id, demand = await requestor.subscribe_demand(demand)
+
+            proposals = await requestor.wait_for_proposals(
+                subscription_id,
+                providers,
+                lambda p: p.properties.get("golem.runtime.name") == "wasmtime",
+            )
+            logger.info("Collected %s proposals", len(proposals))
+            return subscription_id, proposals
+
+        async def negotiate_rejection(
+            requestor, demand, providers, subscription_id, proposals
+        ):
+            counter_providers = []
+            for proposal in proposals:
+                provider = next(p for p in providers if p.address == proposal.issuer_id)
+                logger.info(
+                    "%s Processing proposal from %s", requestor.name, provider.name
+                )
+
+                counter_proposal_id = await requestor.counter_proposal(
+                    subscription_id, demand, proposal
+                )
+                counter_providers.append((counter_proposal_id, provider))
+            return counter_providers
+
+        async def renegotiate(requestor, counter_providers, subscription_id):
+            logger.info("%s: renegotiate()", requestor.name)
+            agreement_providers = []
+            for counter_proposal_id, provider in counter_providers:
+                logger.info(
+                    "%s with %s. p.wait_for_proposal_accepted()",
+                    requestor.name,
+                    provider.name,
+                )
+                # await provider.wait_for_proposal_accepted()
+
+                new_proposals = []
+                collected_offers = await requestor.api.market.collect_offers(
+                    subscription_id
+                )
+                logger.info("collected offers: %s", collected_offers)
+                assert len(collected_offers) == 2
+                assert (
+                    collected_offers[0].reason.message
+                    == "No capacity available. Reached Agreements limit: 1"
+                )
+                new_proposals.append(collected_offers[1].proposal)
+
+                agreement_id = await requestor.create_agreement(new_proposals[0])
+                await requestor.confirm_agreement(agreement_id)
+                await provider.wait_for_agreement_approved()
+                await requestor.wait_for_approval(agreement_id)
+                agreement_providers.append((agreement_id, provider))
+            return agreement_providers
```

Comment:
this should be not empty. assert is missing

Reply:
Removed the iteration so list length is always 1. Left the list format to comply with run() which expects a list.